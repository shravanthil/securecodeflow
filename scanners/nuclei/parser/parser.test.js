// SPDX-FileCopyrightText: the secureCodeBox authors
//
// SPDX-License-Identifier: Apache-2.0

const fs = require("fs");
const util = require("util");
const {
  validateParser,
} = require("@securecodebox/parser-sdk-nodejs/parser-utils");

// eslint-disable-next-line security/detect-non-literal-fs-filename
const readFile = util.promisify(fs.readFile);

const { parse } = require("./parser");

test("parses empty result correctly", async () => {
  const fileContent = 
    await readFile(
      __dirname + "/__testFiles__/empty-test.jsonl",
      {
        encoding: "utf8",
      }
    )

  const findings = await parse(fileContent);
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});

test("parses the example.com result correctly", async () => {
  const fileContent = 
    await readFile(
      __dirname + "/__testFiles__/example-com-test.jsonl",
      {
        encoding: "utf8",
      }
    )

  const findings = await parse(fileContent);
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});

test.skip("parses ftp result correctly", async () => {
  const fileContent = 
    await readFile(
      __dirname + "/__testFiles__/ftp-test.jsonl",
      {
        encoding: "utf8",
      }
    )

  const findings = await parse(fileContent);
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});

test("parses secureCodeBox.io result correctly", async () => {
  const fileContent = 
    await readFile(
      __dirname + "/__testFiles__/secureCodeBox-test.jsonl",
      {
        encoding: "utf8",
      }
    )

  const findings = await parse(fileContent);
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});

test("parses log4shell result correctly", async () => {
  const fileContent = 
    await readFile(
      __dirname + "/__testFiles__/log4shell.jsonl",
      {
        encoding: "utf8",
      }
    )

  const findings = await parse(JSON.parse(fileContent));
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});
