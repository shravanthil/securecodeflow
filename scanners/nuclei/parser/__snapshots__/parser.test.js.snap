// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses empty result correctly 1`] = `Array []`;

exports[`parses ftp result correctly 1`] = `
Array [
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "127.0.0.1",
      "matched": "www.example.com:21",
      "matcher_name": null,
      "metadata": Object {
        "password": "default",
        "username": "anonymous",
      },
      "reference": null,
      "tags": null,
      "timestamp": "2021-08-20T22:00:48.088618+02:00",
      "type": "network",
    },
    "category": "ftp-default-credentials",
    "description": "The name of the nuclei rule which triggered the finding: ftp-default-credentials",
    "location": "www.example.com",
    "name": "FTP Service with default credentials",
    "severity": "HIGH",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "127.0.0.1",
      "matched": "www.example.com:21",
      "matcher_name": null,
      "metadata": Object {
        "password": "stingray",
        "username": "anonymous",
      },
      "reference": null,
      "tags": null,
      "timestamp": "2021-08-20T22:00:48.161349+02:00",
      "type": "network",
    },
    "category": "ftp-default-credentials",
    "description": "The name of the nuclei rule which triggered the finding: ftp-default-credentials",
    "location": "www.example.com",
    "name": "FTP Service with default credentials",
    "severity": "HIGH",
  },
]
`;

exports[`parses log4shell result correctly 1`] = `
Array [
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": Array [
        "vuln-log4j-vulnerable-log4j-74bbf59745-h2kqj",
      ],
      "ip": "10.1.6.107",
      "matched": "http://10.1.6.107:8080/",
      "matcher_name": null,
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:36:21.9627479Z",
      "type": "http",
    },
    "category": "CVE-2021-44228",
    "description": "The name of the nuclei rule which triggered the finding: CVE-2021-44228",
    "location": "http://10.1.6.107:8080",
    "name": "Remote code injection in Log4j",
    "severity": "HIGH",
  },
]
`;

exports[`parses secureCodeBox.io result correctly 1`] = `
Array [
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": Array [
        "TLS13",
      ],
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": null,
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:47.4017488Z",
      "type": "ssl",
    },
    "category": "tls-version",
    "description": "The name of the nuclei rule which triggered the finding: tls-version",
    "location": "https://www.secureCodeBox.io",
    "name": "TLS Version",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io/",
      "matcher_name": "wildcard-no-acac",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:52.1433726Z",
      "type": "http",
    },
    "category": "cors-misconfig",
    "description": "The name of the nuclei rule which triggered the finding: cors-misconfig",
    "location": "https://www.secureCodeBox.io",
    "name": "Basic CORS misconfiguration",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io/",
      "matcher_name": "wildcard-no-acac",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:52.4062729Z",
      "type": "http",
    },
    "category": "cors-misconfig",
    "description": "The name of the nuclei rule which triggered the finding: cors-misconfig",
    "location": "https://www.secureCodeBox.io",
    "name": "Basic CORS misconfiguration",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io/",
      "matcher_name": "wildcard-no-acac",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:52.5676563Z",
      "type": "http",
    },
    "category": "cors-misconfig",
    "description": "The name of the nuclei rule which triggered the finding: cors-misconfig",
    "location": "https://www.secureCodeBox.io",
    "name": "Basic CORS misconfiguration",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": null,
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "github",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:14.8059604Z",
      "type": "dns",
    },
    "category": "cname-service-detection",
    "description": "The name of the nuclei rule which triggered the finding: cname-service-detection",
    "location": "https://www.secureCodeBox.io",
    "name": "cname service detection",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": Array [
        "securecodebox.io",
        "www.securecodebox.io",
      ],
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": null,
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.3341469Z",
      "type": "ssl",
    },
    "category": "ssl-dns-names",
    "description": "The name of the nuclei rule which triggered the finding: ssl-dns-names",
    "location": "https://www.secureCodeBox.io",
    "name": "SSL DNS Names",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "cross-origin-embedder-policy",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.8611628Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://www.secureCodeBox.io",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "clear-site-data",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.8612914Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://www.secureCodeBox.io",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "x-frame-options",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.8613455Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://www.secureCodeBox.io",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "x-content-type-options",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.8614424Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://www.secureCodeBox.io",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "x-permitted-cross-domain-policies",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.8615356Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://www.secureCodeBox.io",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "referrer-policy",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.8616546Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://www.secureCodeBox.io",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "cross-origin-opener-policy",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.8617182Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://www.secureCodeBox.io",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "cross-origin-resource-policy",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.861781Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://www.secureCodeBox.io",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "access-control-allow-credentials",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.8618517Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://www.secureCodeBox.io",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "content-security-policy",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.8618971Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://www.secureCodeBox.io",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "access-control-max-age",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.8619643Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://www.secureCodeBox.io",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "access-control-allow-methods",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.8620127Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://www.secureCodeBox.io",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "185.199.108.153",
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "access-control-expose-headers",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:38.8620724Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://www.secureCodeBox.io",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": null,
      "matched": "https://www.secureCodeBox.io",
      "matcher_name": "nsone",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:57.3588242Z",
      "type": "dns",
    },
    "category": "can-i-take-over-dns-fingerprint",
    "description": "The name of the nuclei rule which triggered the finding: can-i-take-over-dns-fingerprint",
    "location": "https://www.secureCodeBox.io",
    "name": "Can I Take Over DNS - Fingerprint",
    "severity": "INFORMATIONAL",
  },
]
`;

exports[`parses the example.com result correctly 1`] = `
Array [
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "content-security-policy",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6186012Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "x-permitted-cross-domain-policies",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6198316Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "referrer-policy",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6199241Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "clear-site-data",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6199737Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "x-frame-options",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6200435Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "cross-origin-embedder-policy",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6202531Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "cross-origin-opener-policy",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6203117Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "access-control-allow-methods",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6204646Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "cross-origin-resource-policy",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6205434Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "access-control-max-age",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6206546Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "strict-transport-security",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6207415Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "x-content-type-options",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6208071Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "access-control-allow-origin",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6208828Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "access-control-allow-credentials",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6209542Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": null,
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": "access-control-expose-headers",
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:07:19.6210066Z",
      "type": "http",
    },
    "category": "http-missing-security-headers",
    "description": "The name of the nuclei rule which triggered the finding: http-missing-security-headers",
    "location": "https://example.com",
    "name": "HTTP Missing Security Headers",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": Array [
        "example.org",
        "www.example.com",
        "www.example.edu",
        "www.example.net",
        "www.example.org",
        "example.net",
        "example.edu",
        "example.com",
      ],
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": null,
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:04.2602621Z",
      "type": "ssl",
    },
    "category": "ssl-dns-names",
    "description": "The name of the nuclei rule which triggered the finding: ssl-dns-names",
    "location": "https://example.com",
    "name": "SSL DNS Names",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": Array [
        "TLS13",
      ],
      "ip": "93.184.216.34",
      "matched": "https://example.com",
      "matcher_name": null,
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:11.0476586Z",
      "type": "ssl",
    },
    "category": "tls-version",
    "description": "The name of the nuclei rule which triggered the finding: tls-version",
    "location": "https://example.com",
    "name": "TLS Version",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "author": null,
      "extracted_results": Array [
        "31589 8 1 3490A6806D47F17A34C29E2CE80E8A999FFBE4BE",
        "31589 8 2 CDE0D742D6998AA554A92D890F8184C698CFAC8A26FA59875A990C03E576343C",
        "43547 8 1 B6225AB2CC613E0DCA7962BDC2342EA4F1B56083",
        "43547 8 2 615A64233543F66F44D68933625B17497C89A70E858ED76A2145997EDF96A918",
        "31406 8 1 189968811E6EBA862DD6C209F75623D8D9ED9142",
        "31406 8 2 F78CF3344F72137235098ECBBD08947C2C9001C7F6A085A17F518B5D8F6B916D",
      ],
      "ip": null,
      "matched": "https://example.com",
      "matcher_name": null,
      "metadata": null,
      "reference": null,
      "tags": null,
      "timestamp": "2021-12-21T15:08:30.8750541Z",
      "type": "dns",
    },
    "category": "dnssec-detection",
    "description": "The name of the nuclei rule which triggered the finding: dnssec-detection",
    "location": "https://example.com",
    "name": "DNSSEC Detection",
    "severity": "INFORMATIONAL",
  },
]
`;
